### ISSUE TITLE
Customization of table in flatlaf

### ISSUE THREAD

**Issue by VKaddala** [JFormDesigner/FlatLaf#780]
I am adding table customization to my application using flatlaf. I want to achieve something like the following:

<img width="250" alt="image" src="https://github.com/JFormDesigner/FlatLaf/assets/55906470/6d9aab42-381d-44c0-83ab-92325b493a21">

But after doing all property changes, I end up to the below extension:

<img width="155" alt="image" src="https://github.com/JFormDesigner/FlatLaf/assets/55906470/6f9ade5e-6b5f-475a-b6aa-bedd0fd364ae">

Can you please help me to achieve required UI change.

**Comment by VKaddala** [comment]
Can anyone help on this please.

**Comment by DJ-Raven** [comment]
My solution is custom table cell render. I not sure the flatlaf have the properties style

``` java
table.setShowVerticalLines(true);
applyStye(table, 0);
```

``` java
private void applyStye(JTable table, int colIndex) {
    final TableCellRenderer oldCellRender = table.getDefaultRenderer(Boolean.class);
    table.getColumnModel().getColumn(colIndex).setCellRenderer(new DefaultTableCellRenderer() {
        @Override
        public Component getTableCellRendererComponent(JTable table, Object value, boolean isSelected, boolean hasFocus, int row, int column) {
            Component com = oldCellRender.getTableCellRendererComponent(table, value, isSelected, hasFocus, row, column);
            if (isSelected) {
                com.setBackground(UIManager.getColor("Table.selectionBackground"));
            } else {
                if (row % 2 != 0) {
                    com.setBackground(UIManager.getColor("Table.alternateRowColor"));
                } else {
                    com.setBackground(UIManager.getColor("Table.background"));
                }
            }
            return com;
        }
    });
}
```
Here My `FlatLaf.properties`
``` java
Table.alternateRowColor=#F5F5F5

Table.gridColor=#D6D6D6
Table.intercellSpacing=1,1
```
![2023-12-18_164645](https://github.com/JFormDesigner/FlatLaf/assets/58245926/c13870c0-2c3a-4d62-9686-544577514066)

**Comment by DevCharly** [comment]
The problem is that the boolean renderer (see class `JTable.BooleanRenderer`) does not support alternating row colors. This  is IMHO actually a bug in Swing...

Because that boolean renderer is implemented and set in `JTable`, I actually thought that it would be a bad idea to replace it in FlatLaf. But now I found out that other L&Fs (Nimbus, Radiance, ...) do replace that boolean renderer. So I'm going to do the same in FlatLaf... 😄

**Comment by DevCharly** [comment]
fixed in latest `3.3-SNAPSHOT`: https://github.com/JFormDesigner/FlatLaf#snapshots
